packages/api-contract/src/Abi/index.ts

```diff
@@ -162,7 +169,31 @@ export class Abi {
   /**
    * Warning: Unstable API, bound to change
    */
-  public decodeEvent (data: Bytes | Uint8Array): DecodedEvent {
+  public decodeEvent (record: EventRecord): DecodedEvent {
+    switch (this.metadata.version.toString()) {
+      // earlier version are hoisted to v4
+      case '4':
+        return this.#decodeEventV4(record);
+      // Latest
+      default:
+        return this.#decodeEventV5(record);
+    }
+  }
+
+  #decodeEventV5 = (record: EventRecord): DecodedEvent => {
+    const data = record.event.data[1] as Bytes;
+    const signatureTopic = record.topics[0];
+    const event = this.events.find((e) => e.signatureTopic !== undefined && e.signatureTopic === signatureTopic.toHex());
```

Comment:
Some events can be "anonymous", i.e. they have no signature topic. 

In this case we can either:

1. Don't attempt to decode the event and just return the raw bytes, leave it up to the user to do the decoding.
2. Best effort attempt to decode, i.e. try to decode into all anonymous event definitions from the metadata.